src/main/java/ca/on/oicr/gsi/dimsum/controller/MvcExceptionHandler.java

```diff
@@ -14,32 +14,39 @@ public class MvcExceptionHandler {
 
     @Autowired
     private CommonModelAttributeProvider commonModelAttributeProvider;
+
     private static final Logger logger = LoggerFactory.getLogger(MvcExceptionHandler.class);
 
     @ExceptionHandler(ResponseStatusException.class)
     public ModelAndView handleResponseStatusException(ResponseStatusException ex) {
-        ModelAndView mav = new ModelAndView("error");
-        mav.addObject("errorStatus", ex.getStatus().toString());
-        mav.addObject("errorMessage", ex.getReason());
-
-        mav.addObject("dataAgeMinutes", commonModelAttributeProvider.getDataAgeMinutes());
-        mav.addObject("buildVersion", commonModelAttributeProvider.getBuildVersion());
+        ModelAndView mav = prepareErrorModelAndView(ex.getStatus().value(), ex.getReason(), ex);
 
         if (ex.getStatus().is5xxServerError()) {
-            logger.error(ex.getReason(), ex);
+            logger.error("Error Status: " + ex.getStatus() + ". Error Message: " + ex.getReason(),
+                    ex);
+        } else {
+            logger.info("Error Status: " + ex.getStatus() + ". Error Message: " + ex.getReason());
         }
 
-        logger.info("Error Status: " + ex.getStatus());
-        logger.info("Error Message: " + ex.getReason());
-
         return mav;
     }
 
     @ExceptionHandler(Exception.class)
     public ModelAndView handleGeneralException(Exception ex) {
         logger.error("Unexpected error", ex);
+        return prepareErrorModelAndView(500, "Unexpected error", ex);
+    }
+
+    private ModelAndView prepareErrorModelAndView(int errorStatus, String errorMessage,
+            Exception ex) {
         ModelAndView mav = new ModelAndView("error");
-        mav.addObject("errorMessage", "Unexpected error");
+        mav.addObject("errorStatus",
+                errorStatus + (ex instanceof ResponseStatusException
+                        ? " " + ((ResponseStatusException) ex).getStatus().getReasonPhrase()
+                        : ""));
```

Comment:
Add the code and reason phrase as separate properties and display in separate rows on front-end titled "Code" and "Error."

It would also be good to add these both to the title, so it ends up as something like "500 Internal Server Error - Dimsum"